LethalAPI.Bootstrapper/Bootstrapper.cs

```diff
@@ -0,0 +1,419 @@
+            AllocConsole();
+            basePath = Path.Combine(Assembly.GetExecutingAssembly().Location, "../../LethalAPI");
+            configPath = Path.Combine(basePath, "Configs");
+            pluginsPath = Path.Combine(basePath, "Plugins");
+            dependenciesPath = Path.Combine(basePath, "Dependencies");
+
+            if (!Directory.Exists(basePath))
+                Directory.CreateDirectory(basePath);
+
+            if (!Directory.Exists(configPath))
+                Directory.CreateDirectory(configPath);
+
+            if (!Directory.Exists(pluginsPath))
+                Directory.CreateDirectory(pluginsPath);
+
+            if (!Directory.Exists(dependenciesPath))
+                Directory.CreateDirectory(dependenciesPath);
+
+            Loader.LoadMethod = LoadMethod.Doorstop;
+            Loader.Load();
+        }
+
+        /// <summary>
+        /// Loads paths for manual installations.
+        /// </summary>
+        internal static void LoadPathsManual()
+        {
+            PluginLoader.PluginDirectory = pluginsPath;
+            PluginLoader.DependencyDirectory = dependenciesPath;
+            PluginLoader.ConfigDirectory = configPath;
+        }
+
+        /// <summary>
+        /// Logs a message to the output console.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        internal static void Log(string message)
+        {
+            UnityEngine.Debug.Log(message);
+            Console.WriteLine(message);
+        }
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool AllocConsole();
+    }
+}
+#endif
+
+namespace LethalAPI.Bootstrapper
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Reflection;
+
+    using Core.Loader;
+    using HarmonyLib;
+
+    /// <summary>
+    /// The bootstrap loader class.
+    /// </summary>
+    public static class Loader
+    {
+        // ReSharper disable once CollectionNeverQueried.Local
+        private static readonly List<Assembly> Dependencies = new ();
+        private static Assembly baseAssembly = null!;
+
+        /// <summary>
+        /// Gets the <see cref="LoadMethod"/> being used.
+        /// </summary>
+        public static LoadMethod LoadMethod { get; internal set; } = LoadMethod.Manual;
+
+        /// <summary>
+        /// Loads the framework.
+        /// </summary>
+        internal static void Load()
+        {
+            CosturaUtility.Initialize();
+            Log($" [LethalAPI-Bootstrapper] Loading Bootstrapper [{LoadMethod}].");
+            baseAssembly = typeof(Loader).Assembly;
+            LoadPaths();
+            LoadDependencies();
+        }
+
+        private static void Log(string message) => Postfix(message);
+
+        private static void Postfix(string message)
+        {
+#if Melonloader
```

Comment:
I like precompiler variables outdented. I think this is a good conversation to have with the team though. I personally find it more manageable and debuggable generally. Also it can screw with the indentation of other items.

Reply:
by default, all pre-processor is always outdented compare to other thing. And i don't think we should indent it, since this way it make people easier to manage and is fairly more readable compare to when you indent it.

Reply:
Ah, then there's some if macros right after this one that are indentend differently. Please fix!.